frontend/src/general/dialog/Dialog.jsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import MuiDialog from "@material-ui/core/Dialog/Dialog";
+import { DialogTitle } from "@material-ui/core";
+import DialogBody from "./DialogBody";
+import DialogButtons from "./DialogButtons";
+
+export default class Dialog extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false
+    };
+    this.openDialog = this.openDialog.bind(this);
+    this.closeDialog = this.closeDialog.bind(this);
+  }
+
+  openDialog() {
+    return this.setState({ open: true });
+  }
+
+  closeDialog() {
+    return this.setState({ open: false });
+  }
+
+  render() {
+    const cancelButton = {
+      name: "Cancel",
+      onClick: this.closeDialog
+    };
+
+    const { title, actionButton, children } = this.props;
+    const { open } = this.state;
+    return (
+      <MuiDialog
+        open={open}
+        onClose={() => this.closeDialog}
+        aria-labelledby="dialog-title"
+        aria-describedby="dialog-description"
+        id="dialog"
+      >
+        <DialogTitle>{title}</DialogTitle>
+        <DialogBody>{children}</DialogBody>
+        <DialogButtons button1={cancelButton} button2={actionButton} />
```

Comment:
~~my comments didnt post earlier 😫~~
are `button1` and `button2` the best names here? if our designs are consistently putting the cancel button on the left as button1 then it might be clearer to change the names to `cancelButton` and `actionButton` or something

Reply:
Since the dialogButtons are technically their own component, the idea was to leave that flexibility there, though it's unlikely that we will need to reuse it. 
But I do agree the name is a bit weak. Maybe we can rename it to leftButton and rightButton?